server/src/main/java/org/apache/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -271,6 +271,7 @@ private Sink getSink(long timestamp)
           config.getAppendableIndexSpec(),
           config.getMaxRowsInMemory(),
           config.getMaxBytesInMemoryOrDefault(),
+          true,
```

Comment:
i've noticed a few places that aren't wired up to config are using true, so using the new behavior with no way out, is that intentional? this one in particular probably doesn't matter all that much these days (i hope), but i'm less sure about all of them, and it isn't consistent because i see some hard coded false in there too

Reply:
The value of the flag `useMaxMemoryEstimates = true` represents old behaviour.

The hard coding has been done only for the following classes:
- `RealtimePlumber` and related classes (hopefully not used anymore)
- `OnHeapIncrementalIndexBenchmark` (accidentally hardcoded to false, fixing this)

It should NOT be hardcoded to false in any of these cases.